### PR TITLE
[stable/mysqldump] several changes 

### PR DESCRIPTION
**Pull request by monotek**
#### What this PR does / why we need it:
* added gzip compression
* added single db backups
* added split db backups to single backup files
* added ssh upload
* added google storage bucket upload
* added houskeeping (delete old backups in pvc and on remote)
* changed docker image to have google cloud sdk available
* added resource limits config options
* added affinity & tolerations config options
* renamed template files
* changed default backup time to 03:00
* changed app version to reflect helm chart version not mysql client version
* changed labels the helm 2.11 way

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/mysqldump/templates/configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,107 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "mysqldump.fullname" . }}
+  labels:
+    app: {{ template "mysqldump.name" . }}
```
Comment (axdotl) [inline stable/mysqldump/templates/configmap.yaml]: Please make use of the standard labels, see https://docs.helm.sh/chart_best_practices/#standard-labels
(please apply to other templates as well)

File: stable/mysqldump/values.yaml [hunk 2]
```diff
@@ -3,8 +3,8 @@
 # Declare variables to be passed into your templates.
 
 image:
-  repository: mysql
-  tag: "5.7.21"
+  repository: monotek/gcloud-mysql
+  tag: "6"
```
Comment (axdotl) [inline stable/mysqldump/values.yaml]: It's unclear which gcloud sdk and mysql-client version is used here.

File: stable/mysqldump/values.yaml [hunk 3]
```diff
@@ -14,14 +14,21 @@ mysql:
...
 
-## set to `now` to get a one time job, or a cronjob schedule like `0 0 * * *`
-## to get a cronjob.
-schedule: "0/5 * * * *"
-# schedule: now
+# use --all-databases
+allDatabases:
+  enabled: true
+  # creates single backup file with all databases
+  SingleSqlFile: false
```
Comment (axdotl) [inline stable/mysqldump/values.yaml]: Should start with lowercase.

Comment (axdotl) [reply stable/mysqldump/values.yaml]: Take proper name into account, e.g. singleBackupFile or singleDumpFile

File: stable/mysqldump/templates/configmap.yaml [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+    {{- if (.Values.upload.googlestoragebucket.enabled) or (.Values.upload.ssh.enabled) -}}
+    {{ if .Values.upload.ssh.enabled -}}
+    echo "upload files via ssh"
+    rsync -av --delete --exclude=*.state -e 'ssh -o StrictHostKeyChecking=accept-new -o UserKnownHostsFile=/dev/null' ${BACKUP_DIR}/ {{ .Values.upload.ssh.user }}@{{ .Values.upload.ssh.host }}:/{{ .Values.upload.ssh.dir }}
+    rcu=$?
+    {{ end -}}
+
+    {{ if .Values.upload.googlestoragebucket.enabled }}
+    echo "upload files to google storage bucket"
```
Comment (axdotl) [inline stable/mysqldump/templates/configmap.yaml]: Would be good to have the destination logged.

File: stable/mysqldump/templates/configmap.yaml [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    gcloud auth activate-service-account --key-file /root/gcloud/keyfile.json
```
Comment (axdotl) [inline stable/mysqldump/templates/configmap.yaml:81]: Could you add the `--project` flag, to ensure that all operation will be done for the correct project.

File: stable/mysqldump/templates/configmap.yaml [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    {{ else if and (.Values.allDatabases.enabled) (.Values.allDatabases.SingleSqlFile) }}
+    echo "Backing up all databases"
+    MYSQL_DB="alldatabases"
+    mysqldump ${MYSQL_OPTS} -h ${MYSQL_HOST} -P ${MYSQL_PORT} -u ${MYSQL_USERNAME}{{ if .Values.mysql.password }} -p${MYSQL_PASSWORD}{{ end }} --all-databases | gzip > ${BACKUP_DIR}/${TIMESTAMP}_${MYSQL_DB}.sql.gz
+    rc=$?
+    {{- end -}}
+
+    {{- if (.Values.upload.googlestoragebucket.enabled) or (.Values.upload.ssh.enabled) -}}
+    {{ if .Values.upload.ssh.enabled -}}
+    echo "upload files via ssh"
```
Comment (axdotl) [inline stable/mysqldump/templates/configmap.yaml]: Would be good to have the destination logged.

**Review by axdotl — CHANGES_REQUESTED** [review]

File: stable/mysqldump/templates/configmap.yaml (hunk 5 above)
Comment (monotek) [reply stable/mysqldump/templates/configmap.yaml:81]: The project is already stored in the json keyfile.

File: stable/mysqldump/values.yaml (hunk 2 above)
Comment (monotek) [reply stable/mysqldump/values.yaml]: already replaced with chart version

File: stable/mysqldump/templates/configmap.yaml (hunk 1 above)
Comment (monotek) [reply stable/mysqldump/templates/configmap.yaml]: done

File: stable/mysqldump/values.yaml (hunk 3 above)
Comment (monotek) [reply stable/mysqldump/values.yaml]: done

File: stable/mysqldump/templates/configmap.yaml (hunk 4 above)
Comment (monotek) [reply stable/mysqldump/templates/configmap.yaml]: done

File: stable/mysqldump/templates/configmap.yaml (hunk 6 above)
Comment (monotek) [reply stable/mysqldump/templates/configmap.yaml]: done

File: stable/mysqldump/values.yaml (hunk 2 above)
Comment (axdotl) [reply stable/mysqldump/values.yaml]: For the user it is still not clear which gcloudSDK and mysql client is packaged in this docker image. So it's actually an issue of the docker image itself.

**Review by axdotl — APPROVED** [review]

**Comment by monotek** [comment]
pretty please :-)

**Comment by monotek** [comment]
@paulczar 
Could you please review & approve my changes?
If anything is missing let me know.

**Comment by monotek** [comment]
@paulczar 
Could you please review & approve my changes?
If anything is missing let me know.

**Comment by monotek** [comment]
anyone?

**Comment by Scuilion** [comment]
@monotek I've been told that if you want to become a maintainer of this chart you need to create a separate pull request. [EG](https://github.com/helm/charts/pull/9868) Unfortunately, even as a maintainer you cannot merge your own pull request.

Alternatively, if you don't get traction here you can create your own repo and publish it to [Helm Hub](https://www.helm.sh/blog/intro-helm-hub/index.html), which is the way chart management is going.

**Comment by monotek** [comment]
@unguiculus

**Pull request merged**